lib/src/conflicts.rs

```diff
@@ -185,19 +199,37 @@ pub fn materialize_merge_result(
             output.write_all(&content.0)?;
         }
         MergeResult::Conflict(hunks) => {
+            let num_conflicts = hunks
+                .iter()
+                .filter(|hunk| hunk.as_resolved().is_none())
+                .count();
+            let mut conflict_index = 0;
             for hunk in hunks {
                 if let Some(content) = hunk.as_resolved() {
                     output.write_all(&content.0)?;
                 } else {
+                    conflict_index += 1;
                     output.write_all(CONFLICT_START_LINE)?;
+                    output.write_all(
+                        format!(" Conflict {conflict_index} of {num_conflicts}\n").as_bytes(),
+                    )?;
                     let mut add_index = 0;
-                    for left in hunk.removes() {
+                    for (base_index, left) in hunk.removes().enumerate() {
+                        // The vast majority of conflicts one actually tries to
+                        // resolve manually have 1 base.
+                        let base_str = if hunk.removes().len() == 1 {
+                            "base".to_string()
+                        } else {
+                            format!("base #{}", base_index + 1)
+                        };
+
                         let right1 = if let Some(right1) = hunk.get_add(add_index) {
                             right1
                         } else {
                             // If we have no more positive terms, emit the remaining negative
                             // terms as snapshots.
                             output.write_all(CONFLICT_MINUS_LINE)?;
+                            output.write_all(format!(" Contents of {base_str}\n").as_bytes())?;
```

Comment:
These strike me as good places for a tempate render.  Then turning these off becomes as simple as nulling out the template, and the structure of the templated 'ConflictData' object can be extremely educational for anyone looking to understand more about conflicts.  WDYT?

Reply:
I had the same thought, especially if we want to i18n these strings. However, I don't think I want that to be a part of this PR.

IMO, the next thing after this PR would be to figure out where to store and how to pass around a configuration object for the conflict metadata: both what kind of format we want and, if we ever get to that, some information about what each side of the conflict represents (e.g. commit id/description if known).

The first version of that might also not have templates, but once it exists, they would be easy to add.

Reply:
Also, it would seem like we'd need to have a whole separate "templating language" for these. It wouldn't need to be large, but I wonder if there's a way to make creating such a thing less heavyweight.